Automattic/newspack-newsletters
# fix(usage-reports): fix MC usage reports backfill inflated counts

### All Submissions:






### Changes proposed in this Pull Request:



The `get_usage_reports` was written with handling last-day report in mind, and not sufficiently changed in #1541. This PR fixes it, so that prior data is handled properly, and campaign reports are only used to fill in the missing data (and not inflate existing counts in lists activity data). 

### How to test the changes in this Pull Request:

1. Enter `wp shell` and run `array_map(function($r){return implode(',',array_values($r->to_array()));},\Newspack_Newsletters_Mailchimp_Usage_Reports::get_usage_reports( 10 ))` to view last 10 days' data
2. Observe the data is accurately reflecting what you can see in MC

### Other information:
